src/Avalonia.Native/AvaloniaNativeMenuExporter.cs

```diff
@@ -39,7 +39,7 @@ public AvaloniaNativeMenuExporter(IAvaloniaNativeFactory factory)
         public void SetNativeMenu(NativeMenu menu)
         {
             _menu = menu == null ? new NativeMenu() : menu;
-
+            _resetQueued = true; // we've reset the menu to something new, so make sure that a reset is queued
             DoLayoutReset();
```

Comment:
Can we make DoLayoutReset take an argument, `(bool forceUpdate = false)`

and that way we can just call it with true here, and elsewhere code will remain the same?

Reply:
Done. Had to adjust the `Dispatcher.UIThread.Post` in `QueueReset()` too since now `DoLayoutReset` takes a param.